toolchain/parse/typed_nodes.h

```diff
@@ -0,0 +1,781 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#ifndef CARBON_TOOLCHAIN_PARSE_TYPED_NODES_H_
+#define CARBON_TOOLCHAIN_PARSE_TYPED_NODES_H_
+
+#include "toolchain/parse/node_ids.h"
+#include "toolchain/parse/node_kind.h"
+
+namespace Carbon::Parse {
+
+// A pair of a list item and its optional following comma.
```

Comment:
A subheading comment before these three non-parse-node-kind helpers would be helpful.